server/src/main/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregator.java

```diff
@@ -311,9 +311,9 @@ public StarTreeBucketCollector getStarTreeBucketCollector(
             @Override
             public void setSubCollectors() throws IOException {
                 for (Aggregator aggregator : subAggregators) {
-                    if (aggregator instanceof StarTreePreComputeCollector collector) {
-                        this.subCollectors.add(collector.getStarTreeBucketCollector(ctx, starTree, this));
-                    }
+                    this.subCollectors.add(
```

Comment:
Since we would have done the prevalidation, we don't need the `instanceof` check anymore ?

Reply:
Yes. If you find a star-tree query context in search context, it is expected that the entire request should be resolved via star-tree.